src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.List;
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Track.class) {
+            MachineProducer producer = new TrackProducer();
+            return producer.get();
+        }
+        if (type == Bulldozer.class) {
+            MachineProducer producer = new BulldozerProducer();
+            return producer.get();
+        }
+        if (type == Excavator.class) {
+            MachineProducer producer = new ExcavatorProducer();
+            return producer.get();
+        }
+        return null;
```

Comment:
it's not quite good to return null since it can produce NPE, let's throw exception or return empty list here

Reply:
done...added throw Exception